src/cr/cube/cubepart.py

```diff
@@ -123,9 +123,12 @@ def population_fraction(self):
         return self._cube.population_fraction
 
     @lazyproperty
-    def selected_categories_label(self):
+    def selected_ca_category_labels(self):
         """Set of selected categories for all the dimensions."""
-        return {}
+        raise NotImplementedError(
+            "`%s` must implement `.selected_ca_category_labels`"
```

Comment:
I think that this message should be explicit about the fact that the _child_ classes must implement this, and not this class. So if it says that _this_ class must implement it, it's technically wrong.

Reply:
@slobodan-ilic `CubePartition` is never instantiated directly (except in tests) and so it will _always_ be a subclass that triggers this message (which should only be seen by developers anyway).

@ernestoarbitrio just for future reference, I did a little research yesterday and it looks like `type(self).__name__` is preferable over `self.__class__.__name__`, although they both work fine. No need to change this one if you're not in there otherwise, but this pattern comes up from time to time so we should probably use the `type()` version going forward.

Reply:
@ernestoarbitrio why are we saying this should be implemented on all `_CubePartition` subclasses? Isn't it a characteristic that is particular to a `_Slice`?

Reply:
nope, because this property will be called on exporter without cehcking on the partition type beeing a slice a strand or a nub. So we need it in all the subclasses.